### PR TITLE
Add general purpose Service Role

### PR DESCRIPTION
**Pull request by zehicle**
This role can be used by any role that needs to collect service IPs as an attribute.  Very handy for multi-node services (like etdc or kubernetes api or consul-master)

At some point, we could retrofit consul.

REQUIRES that you add an attrib & event declaration in the role.

### CODE REVIEW

**Comment by VictorLowther** [comment]
Can one of the admins verify this patch?

File: rails/app/models/barclamp_rebar/service_role.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  private
+
+  def collect_addresses(nr, *args)
+
+    aname = "#{nr.role.name}-addresses"
+    addresses = []
+
+    d = nr.deployment
```
Comment (galthaus) [inline rails/app/models/barclamp_rebar/service_role.rb]: @VictorLowther, does he need a transaction here?  I think he is good, but not sure.

Comment (zehicle) [inline rails/app/models/barclamp_rebar/service_role.rb]: I considered it - it seemed like a single action, so I did not include it.

Comment (VictorLowther) [inline rails/app/models/barclamp_rebar/service_role.rb]: Wrap it all in a transaction.

Comment (galthaus) [inline rails/app/models/barclamp_rebar/service_role.rb]: Yeah.  I was more thinking that the two on_mumbles were already in a transaction but probably safer just to wrap it

**Pull request merged**